src/packageurl/contrib/url2purl.py

```diff
@@ -265,14 +280,203 @@ def build_sourceforge_purl(uri):
     return purl_from_pattern('sourceforge', sourceforge_pattern, uri)
 
 
-cargo_pattern= (
+cargo_pattern = (
     r"^https?://crates.io/api/v1/crates/"
     r"(?P<name>.+)/(?P<version>.+)"
     r"(\/download)$"
 )
 
 
-#https://crates.io/api/v1/crates/rand/0.7.2/download
+# https://crates.io/api/v1/crates/rand/0.7.2/download
 @purl_router.route(cargo_pattern)
 def build_cargo_purl(uri):
     return purl_from_pattern('cargo', cargo_pattern, uri)
+
+
+github_raw_content_pattern = (
+    r"^https?://raw.githubusercontent.com/"
+    r"(?P<namespace>.+)/(?P<name>.+)/(?P<version>.+)/"
+    r"(?P<subpath>.+)$"
+)
+
+
+@purl_router.route(github_raw_content_pattern)
+def build_github_purl(uri):
+    """
+    Take `uri` as input
+    example uri: https://raw.githubusercontent.com/volatilityfoundation/dwarf2json/master/LICENSE.txt
+    and returns a `PackageURL` object
+    """
+    return purl_from_pattern('github', github_raw_content_pattern, uri)
+
+
+@purl_router.route("https?://api.github\\.com/repos/.*")
+def build_github_purl(uri):
```

Comment:
I am not sure we want this to be the same function name? We probably want them to be context specific, but maybe I am wrong. 

What do you think @pombredanne?

Reply:
Yeah please give me some suggestions on it, I will change it as per suggestions :)

Reply:
`build_github_api_purl` for this specific case. Adapt this pattern to any other cases we have.

Reply:
Cool, it sounds good